src/docfx/lib/moniker/MonikerRangeParser.cs

```diff
@@ -2,20 +2,39 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Generic;
+using System.IO;
 
 namespace Microsoft.Docs.Build
 {
     internal class MonikerRangeParser
     {
         private readonly EvaluatorWithMonikersVisitor _monikersEvaluator;
 
-        public MonikerRangeParser(IEnumerable<Moniker> monikers)
+        public MonikerRangeParser(MonikerDefinitionModel monikerDefinition)
         {
-            _monikersEvaluator = new EvaluatorWithMonikersVisitor(monikers);
+            if (monikerDefinition == null)
```

Comment:
Can the input be null here?

Reply:
I think not. We can `Debug.Assert` here.

---
In reply to: [230628458](https://github.com/dotnet/docfx/pull/3575#discussion_r230628458) [](ancestors = 230628458)

Reply:
monikerDefinition is required only when `monikerRange` is used?

---
In reply to: [230629457](https://github.com/dotnet/docfx/pull/3575#discussion_r230629457) [](ancestors = 230629457,230628458)

Reply:
changed

---
In reply to: [230630279](https://github.com/dotnet/docfx/pull/3575#discussion_r230630279) [](ancestors = 230630279,230629457,230628458)